File: frontend/web/src/api/coordinatorDistribution.ts
/*
 * GET /api/coordinator/distribution — агрегат для admin.html view-distribution.
 * Координатор видит ВСЕ активные проекты + общий пул нераспределённых.
 *
 * Контракт зеркалит CoordinatorDistributionResponse в Go-бэке.
 */

import { apiFetch } from './client';
import type {
  MentorDistributionTeam,
} from './mentorDistribution';
import type { ApplicationStatus } from './applications';
import type { ProjectStatus } from './projects';

export interface CoordinatorDistributionMentor {
  id: number;
  firstName: string;
  lastName: string;
}

export interface CoordinatorDistributionProject {
  id: number;
  title: string;
  status: ProjectStatus;
  company: string;
  mentor?: CoordinatorDistributionMentor | null;
  teamSizeMin: number;
  teamSizeMax: number;
  numTeams: number;
  teams: MentorDistributionTeam[];
}

export interface CoordinatorPoolPriority {
  applicationId: number;
  projectId: number;
  projectTitle: string;
  priority: number;
  status: ApplicationStatus;
}

export interface CoordinatorPoolStudent {
  studentId: number;
  firstName: string;
  lastName: string;
  course: number;
  group: string;
  gpa: number;
  priorities: CoordinatorPoolPriority[];
}

export interface CoordinatorDistributionResponse {
  deadline?: string;
  projects: CoordinatorDistributionProject[];
  pool: CoordinatorPoolStudent[];
}

export function getCoordinatorDistribution(): Promise<CoordinatorDistributionResponse> {
  return apiFetch<CoordinatorDistributionResponse>('/coordinator/distribution');
}

File: backend/project-service/internal/models/coordinator_distribution.go
package models

// CoordinatorDistributionResponse — агрегат для GET /api/coordinator/distribution
// (admin.html view-distribution). Координатор видит ВСЕ активные проекты с
// командами и общий пул нераспределённых студентов.
//
// Отличия от MentorDistributionResponse:
//   - проекты не фильтруются по mentor_id
//   - возвращаются ВСЕ команды (как launched, так и нет), потому что
//     координатор может перемещать студентов и в запущенные команды (для
//     edge case переназначения)
//   - пул один общий, а не по-проектный — admin.html показывает
//     «Нераспределённые (N)» с плоским списком
type CoordinatorDistributionResponse struct {
	Deadline string                          `json:"deadline,omitempty"`
	Projects []CoordinatorDistributionProject `json:"projects"`
	Pool     []CoordinatorPoolStudent         `json:"pool"`
}

type CoordinatorDistributionProject struct {
	ID          int                            `json:"id"`
	Title       string                         `json:"title"`
	Status      ProjectStatus                  `json:"status"`
	Company     string                         `json:"company"`
	Mentor      *UserSummary                   `json:"mentor,omitempty"`
	TeamSizeMin int                            `json:"teamSizeMin"`
	TeamSizeMax int                            `json:"teamSizeMax"`
	NumTeams    int                            `json:"numTeams"`
	Teams       []MentorDistributionTeam       `json:"teams"`
}

// CoordinatorPoolStudent — чип в боковой панели «Нераспределённые». Включает
// все приоритеты студента, чтобы в drawer'е можно было их показать.
type CoordinatorPoolStudent struct {
	StudentID  int                       `json:"studentId"`
	FirstName  string                    `json:"firstName"`
	LastName   string                    `json:"lastName"`
	Course     int                       `json:"course"`
	Group      string                    `json:"group"`
	GPA        float64                   `json:"gpa"`
	Priorities []CoordinatorPoolPriority `json:"priorities"`
}

type CoordinatorPoolPriority struct {
	ApplicationID int               `json:"applicationId"`
	ProjectID     int               `json:"projectId"`
	ProjectTitle  string            `json:"projectTitle"`
	Priority      int               `json:"priority"`
	Status        ApplicationStatus `json:"status"`
}
